oeps/oep-0011.rst

```diff
@@ -0,0 +1,412 @@
+   web applications follow best practices for user experience, theming,
+   and accessibility.
+
+   You can learn more about building user interfaces with the pattern
+   library in the Developer's Guide chapter `Developing User Interfaces`_.
+
+4. **Front-ends should consume and extract reusable UI components**
+
+   Rationale: when developing new applications, developers should prefer
+   reuse of existing components over building custom front end logic.
+   If a new component is required, it should be built as a new reusable
+   addition to the `edX UI Toolkit`_, based upon patterns from the
+   `edX Pattern Library`_. This ensures that future development can
+   reuse this component, while also standardizing how it behaves
+   for the user in terms of user interface, accessibility, theming
+   and so on.
+
+6. **Front-ends should manage history consistently**
+
+   Rationale: it is important that rich-client applications provide a
+   consistent history to their users. This includes both the ability to
+   use the back button in a natural way, and to be able to use URLs to
+   deep link to any part of the application. History should be updated
+   for every action that feels to the user that they are on a new page.
+
+7. **Features should work equally well on mobile and web platforms**
+
+   Rationale: Open edX provides iOS and Android applications in addition
+   to its web applications, and so new learner features need to be implemented
+   for all three platforms. The simplest way to achieve feature parity is to
+   provide a responsive web experience that can be used on phones and tablets
+   as well as on desktops and laptops. The `edx Pattern Library`_ helps with
+   this by providing a responsive grid system along with components that
+   work equally well on mobile and web.
+
+8. **Designs should be fully accessible**
+
+   Rationale: Open edX measures and evaluates accessibility using the World
+   Wide Web Consortium’s `Web Content Accessibility Guidelines (WCAG) 2.0`_.
+   All features merged into edX repositories are expected to conform to
+   Level AA of this specification and to satisfy the requirements outlined in
+   the `edX Website Accessibility Policy`_.
+
+   You can learn more about developing accessible applications with
+   the Developer's Guide chapter `edX Accessibility Guidelines`_.
+
+9. **Front end applications should support localization**
+
+   Rationale: Open edX is used by learners all over the world, and so
+   it is critical that all new functionality support localization. This means
+   that every string in an application should be marked for translation,
+   and it also includes ensuring that all new user interfaces will switch to
+   right-to-left mode when the language requires it.
+
+   You can learn more about localization with the Developer's Guide
+   chapter `edX Internationalization`_.
+
+10. **Front end implementations should support theming**
```

Comment:
Theming is not a requirement that we have had before-- we never discussed this in the capa feedback work, for example. In my opinion, this is much lower in priority compared to the other items in this list, and it does not have company buy-in at this time.